building tests/subsys/jwt failed on disco_l475_iot1  with twister

When running the twister -T tests/subsys/jwt/  on -p disco_l475_iot1 -
the build fails due to "FAILED: zephyr/kernel/CMakeFiles/kernel.dir/idle.c.obj"

Please also mention any information which could help others to understand the problem you're facing:
 - target platform disco_l475_iot1
 - regression   after the merge of https://github.com/zephyrproject-rtos/zephyr/pull/50508  (commit d823f88e3b3478050f16431526ebdc550152a0af)
 
**To Reproduce**
Steps to reproduce the behavior:
1. twister -b -p disco_l475_iot1 -T tests/subsys/jwt/
3. build error


**Impact**
CI and test campaign failure  


**Logs and console output**
```
./zephyr/kernel/idle.c: In function 'idle':
./zephyr/kernel/idle.c:51:18: error: this 'for' clause does not guard... [-Werror=misleading-indentation]
   51 |                     !IS_ENABLED(CONFIG_SCHED_IPI_SUPPORTED)) {
      |                  ^  
./zephyr/kernel/idle.c:53:32: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the 'for'
   53 |                                 ;
      |                         ^       ~             

cc1: all warnings being treated as errors
```



**Environment (please complete the following information):**
 - OS: ubuntu
 - Toolchain SDK, 0.15.0
 - Commit SHA 25f14d7092fcdc54de0e144d36da549925531708

**Additional context**
`$ west build -p auto -b disco_l475_iot1 tests/subsys/jwt/`
is successful

--- a/kernel/idle.c
+++ b/kernel/idle.c
@@ -47,10 +47,10 @@ void idle(void *unused1, void *unused2, void *unused3)
 		 * lock and/or timer driver.  This is intended as a
 		 * fallback configuration for new platform bringup.
 		 */
-		if (IS_ENABLED(CONFIG_SMP) &&
-		    !IS_ENABLED(CONFIG_SCHED_IPI_SUPPORTED)) {
-			for (volatile int i = 0; i < 100000; i++)
-				;
+		if (IS_ENABLED(CONFIG_SMP) && !IS_ENABLED(CONFIG_SCHED_IPI_SUPPORTED)) {
+			for (volatile int i = 0; i < 100000; i++) {
+				/* Empty loop */
+			}
 			z_swap_unlocked();
 		}
 

